### PR TITLE
Hooked up the IdentifyObject event and returned a debug string.

### PR DESCRIPTION
**Pull request by OptimShi**
Also added the flags for when we get to the point we are passing actual data.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ### 2017-04-26
 [OptimShi]
+* Hooked up the "ExaminationQueuePop" in Player.cs to fire.
+* Added IdentifyResponseFlags, for when we start to send actual data.
+* Added a basic debug string to any ID. It simply returns the objects GUID.
+
+### 2017-04-26
```
Comment (LtRipley36706) [inline changelog.md]: combine

Comment (LtRipley36706) [inline changelog.md]: Wait, sorry, the date needs to be changed not combined. My bad.

File: Source/ACE/Network/GameEvent/Events/GameEventIdentifyObjectResponse.cs [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+            Writer.Write((uint)flags); // Flags
+            Writer.Write(1u); // Success bool
+
+            // Write a simple debug strings. Thanks to Pea for this idea.
+            Writer.Write(1u + (1 << 16)); // One String, + table size
+            Writer.Write(16u); // Long Description String
+            Writer.WriteString16L("This item has a guid of " + objectID.ToString() + " (0x" + objectID.ToString("X") + ")");
         }
     }
 }
```
Comment (LtRipley36706) [inline Source/ACE/Network/GameEvent/Events/GameEventIdentifyObjectResponse.cs]: end on new line please

File: Source/ACE/Network/Enum/IdentifyResponseFlags.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        ArmorProfile = 0x80,
+        CreatureProfile = 0x100,
+        WeaponProfile = 0x20,
+        HookProfile = 0x40,
+        ArmorEnchantmentBitfield = 0x200,
+        WeaponEnchantmentBitfield = 0x800,
+        ResistEnchantmentBitfield = 0x400,
+        ArmorLevels = 0x4000,
+    }
+}
```
Comment (LtRipley36706) [inline Source/ACE/Network/Enum/IdentifyResponseFlags.cs]: new line here as well

File: Source/ACE/Network/GameEvent/Events/GameEventIdentifyObjectResponse.cs (hunk 2 above)
Comment (OptimShi) [inline Source/ACE/Network/GameEvent/Events/GameEventIdentifyObjectResponse.cs]: Is there a way to add this to the StyleCop settings? It didn't throw up any alert on this. I have made a mental note, however, for the future.

**Comment by OptimShi** [comment]
Per discussion, I modified this to only spit out debug info on "DebugObjects" (as opposed to proper WorldObjects, which as of now are players, life stones and portals.) Also added the WeenieClassId for further debugging fun (which required modifying how this works a little bit, but all for the best)

File: Source/ACE/Entity/Landblock.cs [hunk 4]
```diff
@@ -477,6 +477,13 @@ public void UseTime()
                     QueuedGameAction action = player.ActionQueuePop();
                     if (action != null)
                         HandleGameAction(action, player);
+                    else
```
Comment (Mogwai-AC) [inline Source/ACE/Entity/Landblock.cs]: Is there a reason we'd only process the Examination Queue when there wasn't a pending action?  I don't get the point of putting this in an else block.

File: Source/ACE/Entity/Landblock.cs [hunk 5]
```diff
@@ -596,6 +603,18 @@ private void HandleGameAction(QueuedGameAction action, Player player)
                         HandleSoundEvent(obj, soundEffect);
                         break;
                     }
+                case GameActionType.IdentifyObject:
+                    {
+                        var g = new ObjectGuid(action.ObjectId);
```
Comment (Mogwai-AC) [inline Source/ACE/Entity/Landblock.cs]: we know the retail servers throttled this, we might want to put a "TODO" comment here to that effect.

File: Source/ACE/Entity/Landblock.cs (hunk 4 above)
Comment (OptimShi) [inline Source/ACE/Entity/Landblock.cs]: It was a way of giving lower priority to the examine request and also ensure only one action is performed per "tick" so as not to try and do too much at once. Can just as easily modify it to try and perform both an GameAction and an ExamineAction, if any, though.

**Pull request merged**